src/hooks/usePlayground.js

```diff
@@ -53,29 +59,32 @@ function reducer(state, action) {
   }
 }
 
-function usePlayground(props) {
-  let { markup, query, onChange, instanceId, rootNode } = props || {};
+const onStateChange = ({ markup, query, result }) => {
+  globalState.save({ markup, query });
+  globalState.updateTitle(result?.expression?.expression);
+};
+
+const usePlayground = () => {
+  let { markup, query } = globalState.load();
 
   if (!markup && !query) {
     markup = defaultValues.markup;
     query = defaultValues.query;
   }
 
-  const result = parser.parse({ rootNode, markup, query, cacheId: instanceId });
+  const result = parser.parse({ markup, query });
```

Comment:
And the `cacheId` can be used as optimization.

Reply:
`rootNode` & `instanceId` are props from `usePlayground`, but neither `Embedder` & `Playground` (which are the only users of `usePlayground`) are passing those, so that's why I removed them from `props`.
I also removed them as arguments to other functions, since they'll always be `undefined` anyways.

Reply:
Aah, yes. You're right. The chrome extension doesn't use the hook. It calls parser directly.

Reply:
Regarding instanceId, I think it's better to set it to a value, than to delete it. It's an easy perf win.

Reply:
Just setting an `uuid` could help out I think?

Reply:
Nah, just name it `playground` in the Playground component, and `embed` in the Embedded.